### PR TITLE
feat(acr): azure container registry plugin

### PR DESCRIPTION
**Pull request by invincibleJai**
**Tracks:** https://github.com/janus-idp/backstage-plugins/issues/216

This is based on https://github.com/janus-idp/backstage-plugins/pull/313

**Description:** Azure container registry plugin allows Backstage users to view details of an image related to a catalog resource deployed from your Azure Container Registry.

**Screenshots/Gif:**

Light Theme:
image

Dark Theme:
image

API Failure: 
image

**Tests:**
image

**TODO:**

- [x] Handle open review comments on https://github.com/janus-idp/backstage-plugins/pull/313
- [x] Refactor code and clean up
- [x] Error handling
- [x] Add Dev Setup
- [x] Add Tests

### CODE REVIEW

File: plugins/acr/dev/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            new MockAzureContainerRegistryApiClient(mockAcrTagsData),
+          ],
+        ]}
+      >
+        <EntityProvider entity={mockEntity}>
+          <AcrPage />
+        </EntityProvider>
+      </TestApiProvider>
+    ),
+    title: 'Root Page',
```
Comment (debsmita1) [inline plugins/acr/dev/index.tsx:47]: update the title

**Comment by debsmita1** [comment]
The registry should be a link
acr
Screenshot 2023-07-05 at 12 04 30 AM

quay
Screenshot 2023-07-05 at 12 04 13 AM

**Comment by invincibleJai** [comment]
> The registry repo should be a link acr Screenshot 2023-07-05 at 12 04 30 AM
> 
> quay Screenshot 2023-07-05 at 12 04 13 AM

am not sure why they kept in dev mode but in actual it is not, check below

image

**Comment by debsmita1** [comment]
The sort on the manifest column is not working. Try using the column `render` instead of modifying the Table data.

File: plugins/acr/src/components/AzureContainerRegistry/AzureContainerRegistry.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { Progress, Table } from '@backstage/core-components';
+import { useApi } from '@backstage/core-plugin-api';
+
+import { AzureContainerRegistryApiRef } from '../../api';
+import { Tag, TagRow } from '../../types';
+import { formatDate } from '../utils';
+import { ErrorReport } from './ErrorReport';
+import { columns, useStyles } from './tableHeading';
+
+export const AzureContainerRegistry = ({ image }: { image: string }) => {
```
Comment (divyanshiGupta) [inline plugins/acr/src/components/AzureContainerRegistry/AzureContainerRegistry.tsx:13]: nit:
Can you create separate type for component props?

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Comment by tumido** [comment]
The SonarCloud quality gate can be solved later with the slow refactoring of shared components into `@janus-idp/shared-react`. This can be a separate task.

**Review by tumido — APPROVED** [review]

**Pull request merged**